x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -587,6 +605,15 @@ private LogicalPlan resolveEnrich(Enrich enrich, List<Attribute> childrenOutput)
             }
             return enrich;
         }
+
+        private DataType[] allowedEnrichTypes(String matchType) {
```

Comment:
nit:  This can be a static map from match type to set of supported data types.

Reply:
Agreed. Since this was not a hot path I did not both with that, but could easily refine it. I did want your opinion on the rules though, since the old code only supported keyword and ip for `match`, and I've maintained that rule, but think it might already work for the other types supported in `range`.

Reply:
Reduced the strictness on non-geo types for match/range.

This also allows keywords to be used in ip_range matches, which was the previous behaviour. However, this also means that invalid IP strings will throw errors. I added a test asserting on that error, and think this matches previous behaviour, but wonder if we should instead return null and set a warning.